# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestCallWorkerThread.cpp
# Commit: 7a33c8b73972
# Full Hash: 7a33c8b7397200d89a0d3904669777e60550c445
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1968812
# File Overlap Count: 1
# Description:
#   Bug 1968812 - Remove the CallWorker thread and use TaskQueue::Observer instead. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D252686
# ==============================================================================

diff -r a3ad3abc4d02 -r 7a33c8b73972 dom/media/gtest/TestCallWorkerThread.cpp
--- a/dom/media/gtest/TestCallWorkerThread.cpp	Wed Jun 11 13:37:22 2025 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,73 +0,0 @@
-/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-
-#include "CallWorkerThread.h"
-
-#include "gmock/gmock.h"
-#include "gtest/gtest.h"
-
-using testing::InSequence;
-using testing::MockFunction;
-
-namespace mozilla {
-
-RefPtr<CallWorkerThread> MakeTestCallWorkerThread() {
-  return new CallWorkerThread(
-      MakeUnique<WebrtcTaskQueueWrapper<DeletionPolicy::NonBlocking>>(
-          do_AddRef(GetCurrentSerialEventTarget()), "TestCallWorkerThread"_ns,
-          true));
-}
-
-TEST(TestCallWorkerThread, TestCurrent)
-{
-  auto wt = MakeTestCallWorkerThread();
-
-  MockFunction<void(int)> checkpoint;
-  {
-    InSequence s;
-    EXPECT_CALL(checkpoint, Call(1));
-    EXPECT_CALL(checkpoint, Call(2));
-  }
-
-  wt->Dispatch(NS_NewRunnableFunction(__func__, [&] {
-    checkpoint.Call(2);
-    EXPECT_TRUE(wt->IsCurrentThreadIn());
-  }));
-  checkpoint.Call(1);
-  NS_ProcessPendingEvents(nullptr);
-}
-
-TEST(TestCallWorkerThread, TestDispatchDirectTask)
-{
-  auto wt = MakeTestCallWorkerThread();
-
-  MockFunction<void(int)> checkpoint;
-  {
-    InSequence s;
-    EXPECT_CALL(checkpoint, Call(1));
-    EXPECT_CALL(checkpoint, Call(2));
-    EXPECT_CALL(checkpoint, Call(3));
-    EXPECT_CALL(checkpoint, Call(4));
-  }
-
-  wt->Dispatch(NS_NewRunnableFunction(__func__, [&] {
-    checkpoint.Call(2);
-    AbstractThread::DispatchDirectTask(
-        NS_NewRunnableFunction("TestDispatchDirectTask Inner", [&] {
-          checkpoint.Call(3);
-          EXPECT_TRUE(wt->IsCurrentThreadIn());
-        }));
-  }));
-
-  wt->Dispatch(NS_NewRunnableFunction(__func__, [&] {
-    checkpoint.Call(4);
-    EXPECT_TRUE(wt->IsCurrentThreadIn());
-  }));
-  checkpoint.Call(1);
-  NS_ProcessPendingEvents(nullptr);
-}
-
-}  // namespace mozilla